### PR TITLE
Enhance Elastic extension for configurable ports

### PR DESCRIPTION
**Pull request by brendonparker**
Proposal for #937 

Sample usage (tye.yaml)
```
name: tye_hello_world
registry: localhost:5000
extensions:
- name: elastic
  logPath: ./.logs
  port: 11000
services:
- name: frontend
  project: frontend/frontend.csproj
- name: backend
  project: backend/backend.csproj
  ```
  
  This allows you to work around scenarios where the default ports (`5601` and `9200`) are locked down by an administrator.

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/tye?pullRequest=938) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/tye?pullRequest=938) before we can accept your contribution.<br/><br/>:x: brendonparker [sign now](https://cla.dotnetfoundation.org/dotnet/tye?pullRequest=938)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/tye?pullRequest=938) it.</sub>

File: src/Microsoft.Tye.Extensions/Elastic/ElasticStackExtension.cs [hunk 1]
```diff
@@ -105,5 +110,14 @@ public override Task ProcessAsync(ExtensionContext context, ExtensionConfigurati
 
             return Task.CompletedTask;
         }
+
+        private static int? GetIntValueFromConfigData(ExtensionConfiguration config, string key, int defaultValue)
+        {
+            if (config.Data.TryGetValue(key, out var obj) && int.TryParse(obj?.ToString(), out int parsedValue))
```
Comment (davidfowl) [inline src/Microsoft.Tye.Extensions/Elastic/ElasticStackExtension.cs:116]:
```suggestion
            if (config.Data.TryGetValue(key, out var obj) && obj is not null && int.TryParse(obj.ToString(), out int parsedValue))
```

**Comment by brendonparker** [comment]
Happy to just close this PR if its not valuable/needed.

**Comment by brendonparker** [comment]
Closing this as I no longer have a need for it, and based on how long its been around, neither does anyone else.

**Pull request closed without merging**